.chloggen/component-refactor-datatype-2.yaml

```diff
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: componentprofiles
```

Comment:
I feel like we can do this one in one go since this is quite experimental yet

Reply:
We end up needing `DataTypeProfiles` to support the deprecation process in `service`.  The deprecated `Config.Validate` for example needs it.

We could update that method to convert from `SignalProfiles` to a `DataType`, but I'm not sure thats easier.

Reply:
Fair